main/src/ca/uwaterloo/flix/language/phase/unification/Substitution.scala

```diff
@@ -125,7 +126,8 @@ case class Substitution(m: Map[Symbol.KindedTypeVarSym, Type]) {
       this
     } else {
       Substitution(
-        this.m ++ that.m.filter(kv => !this.m.contains(kv._1))
+        this.m ++ that.m.filter(kv => !(this.m.contains(kv._1) || this.trueVars.contains(kv._1))),
+        this.trueVars ++ that.trueVars.filter(v => !(this.m.contains(v) || this.trueVars.contains(v)))
```

Comment:
I don't think the filter is needed here? Because both left and right map to the same (TRUE).

Reply:
this.trueVars.contains(v) is redundant, but we should filter out those that are already in this.m, since otherwise some variables might be in both m and trueVars in the result